tests/src/Traits/ScreenShotTrait.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace Drupal\joinup\Traits;
+
+use Behat\Mink\Driver\Selenium2Driver;
+use Behat\Behat\Context\Environment\InitializedContextEnvironment;
+use Behat\Behat\Hook\Scope\AfterStepScope;
+use Behat\Mink\Exception\DriverException;
+
+/**
+ * Trait ScreenShot forked from https://github.com/nuvoleweb/drupal-behat.
+ *
+ * @see https://github.com/nuvoleweb/drupal-behat
+ */
+trait ScreenShotTrait {
+
+  /**
+   * Saves a screen-shot under a given name.
+   *
+   * @param string $name
+   *   The file name.
+   *
+   * @Then (I )take a screenshot :name
+   */
+  public function takeScreenshot($name = NULL) {
+    $file_name = sys_get_temp_dir() . DIRECTORY_SEPARATOR . $name;
+    $message = "Screenshot created in @file_name";
+    $this->createScreenshot($file_name, $message, FALSE);
+  }
+
+  /**
+   * Saves a screen-shot under a generated name.
+   *
+   * @Then (I )take a screenshot
+   */
+  public function takeScreenshotUnnamed() {
+    $file_name = sys_get_temp_dir() . DIRECTORY_SEPARATOR . 'behat-screenshot';
```

Comment:
I really don't know. As I added in the trait docblock, this code was stolen. I didn't touch it except the class namespace. I changed the phpdoc. But I didn't used this method. You'll get automatic screenshots with `takeScreenshotAfterFailedStep()`. We can improve this code for sure but is somehow out-of-scope. I agree that adding this trait is also out-of-scope.